test/integration/fapi-export-policy.int.c

```diff
@@ -0,0 +1,152 @@
+read_policy(FAPI_CONTEXT *context, char *policy_name)
+{
+    FILE *stream = NULL;
+    long policy_size;
+    char *json_policy = NULL;
+    char policy_file[1024];
+
+    if (snprintf(&policy_file[0], 1023, TOP_SOURCEDIR "/test/data/fapi/%s.json", policy_name) < 0)
+        return NULL;
+
+    stream = fopen(policy_file, "r");
+    if (!stream) {
+        LOG_ERROR("File %s does not exist", policy_file);
+        return NULL;
+    }
+    fseek(stream, 0L, SEEK_END);
+    policy_size = ftell(stream);
+    fclose(stream);
+    json_policy = malloc(policy_size + 1);
+    stream = fopen(policy_file, "r");
+    ssize_t ret = read(fileno(stream), json_policy, policy_size);
+    if (ret != policy_size) {
+        LOG_ERROR("IO error %s.", policy_file);
+        return NULL;
+    }
+    json_policy[policy_size] = '\0';
+    return json_policy;
+}
+
+/** Test the FAPI key signing with PolicyAuthorizeNV.
+ *
+ * Tested FAPI commands:
+ *  - Fapi_Provision()
+ *  - Fapi_Import()
+ *  - Fapi_ExportPolicy()
+ *
+ * @param[in,out] context The FAPI_CONTEXT.
+ * @retval EXIT_FAILURE
+ * @retval EXIT_SUCCESS
+ */
+int
+test_fapi_export_policy(FAPI_CONTEXT *context)
+{
+    TSS2_RC r;
+    size_t i;
+
+    static char*  policies[] = {
+        "/policy/pol_action",
+        "/policy/pol_pcr16_0_ecc_authorized",
+        "/policy/pol_authorize_ecc_pem",
+        "/policy/pol_nv_counter",
+        "/policy/pol_authorize_rsa_pem",
+        "/policy/pol_locality",
+        "/policy/pol_nv_change_auth",
+        "/policy/pol_password",
+        "/policy/pol_pcr16_0_or",
+        "/policy/pol_physical_presence",
+        "/policy/pol_secret",
+        "/policy/pol_authorize",
+        "/policy/pol_authorize_nv",
+        "/policy/pol_auth_value",
+        "/policy/pol_command_code",
+        "/policy/pol_duplicate",
+        "/policy/pol_pcr16_0",
+        "/policy/pol_nv",
+        "/policy/pol_authorize_outer",
+        "/policy/pol_countertimer",
+        "/policy/pol_cphash",
+        "/policy/pol_name_hash",
+        "/policy/pol_nv_written",
+        "/policy/pol_pcr16_0_fail",
+        "/policy/pol_pcr16_read",
+        "/policy/pol_pcr8_0",
+        "/policy/pol_signed_ecc",
+        "/policy/pol_signed",
+    };
+
+    char *json_policy = NULL;
+    char *policy = NULL;
+
+    r = Fapi_Provision(context, NULL, NULL, NULL);
+    goto_if_error(r, "Error Fapi_Provision", error);
+
+    for (i = 0; i < sizeof(policies) / sizeof(policies[0]); i++) {
+        fprintf(stderr, "\nTest policy: %s\n",  policies[i]);
+        json_policy = read_policy(context, policies[i]);
+        if (!json_policy)
+            goto error;
+
+        r = Fapi_Import(context, policies[i], json_policy);
+        goto_if_error(r, "Error Fapi_Import", error);
+
+        policy = NULL;
+        r = Fapi_ExportPolicy(context, policies[i], &policy);
+        goto_if_error(r, "Error Fapi_ExportPolicy", error);
+        assert(policy != NULL);
+        assert(strlen(policy) > ASSERT_SIZE);
```

Comment:
IMHO this should contain a check for the policyDigest to be correct (i.e. strcmp() against a known good value), since that the reason of this test.